conan/internal/api/detect_api.py

```diff
@@ -568,6 +569,9 @@ def default_compiler_version(compiler, version):
     of the minor or patch digits, that do not affect binary compatibility
     """
     output = ConanOutput(scope="detect_api")
+    if not version:
+        raise ConanException(
+            f"No version provided to default_compiler_version for {compiler} compiler")
```

Comment:
This exception doesn't have context as the ``output`` that at least reads ``detect_api``.
I would add something:
```suggestion
            f"No version provided to 'detect_api.default_compiler_version()' for {compiler} compiler")
```

The error might be less evident if profile is used in ``conan install``. Maybe even better if we could check there is an error message like: "Error in profile 'myprofile': No version ..."

Reply:
I see in @perseoGI comment that the profile part is already being added 👍